tests/FilesystemsTest.php

```diff
@@ -0,0 +1,150 @@
+                    ->getMongoDB()
+                    ->selectGridFSBucket(),
+            ],
+        ];
+    }
+
+    #[DataProvider('provideValidOptions')]
+    public function testValidOptions(array $options)
+    {
+        // Service used by "bucket-service"
+        $this->app->singleton('bucket', static fn (Application $app) => $app['db']
+            ->connection('mongodb')
+            ->getMongoDB()
+            ->selectGridFSBucket());
+
+        $this->app['config']->set('filesystems.disks.' . $this->dataName(), $options);
+
+        $disk = Storage::disk($this->dataName());
+        $disk->put($filename = $this->dataName() . '.txt', $value = microtime());
+        $this->assertEquals($value, $disk->get($filename), 'File saved');
+    }
+
+    public static function provideInvalidOptions(): Generator
+    {
+        yield 'not-mongodb-connection' => [
+            ['driver' => 'gridfs', 'connection' => 'sqlite'],
+            'The database connection "sqlite" does not use the "mongodb" driver.',
+        ];
+
+        yield 'factory-not-bucket' => [
+            ['driver' => 'gridfs', 'bucket' => static fn () => new stdClass()],
+            'Unexpected value for GridFS "bucket" configuration. Expecting "MongoDB\GridFS\Bucket". Got "stdClass"',
+        ];
+
+        yield 'service-not-bucket' => [
+            ['driver' => 'gridfs', 'bucket' => 'bucket'],
+            'Unexpected value for GridFS "bucket" configuration. Expecting "MongoDB\GridFS\Bucket". Got "stdClass"',
+        ];
+    }
+
+    #[DataProvider('provideInvalidOptions')]
+    public function testInvalidOptions(array $options, string $message)
+    {
+        // Service used by "service-not-bucket"
+        $this->app->singleton('bucket', static fn () => new stdClass());
+
+        $this->app['config']->set('filesystems.disks.' . $this->dataName(), $options);
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage($message);
+
+        Storage::disk($this->dataName());
+    }
+
+    public function testReadOnlyAndThrowOption()
+    {
+        $this->app['config']->set('filesystems.disks.gridfs-readonly', [
+            'driver' => 'gridfs',
+            'connection' => 'mongodb',
+            // Use ReadOnlyAdapter
+            'read-only' => true,
+            // Throw exceptions
+            'throw' => true,
+        ]);
+
+        $disk = Storage::disk('gridfs-readonly');
+
+        $this->expectException(UnableToWriteFile::class);
+        $this->expectExceptionMessage('This is a readonly adapter.');
+
+        $disk->put('file.txt', '');
+    }
+
+    public function testPrefix()
+    {
+        $this->app['config']->set('filesystems.disks.gridfs-prefix', [
+            'driver' => 'gridfs',
+            'connection' => 'mongodb',
+            'prefix' => 'foo/bar/',
+        ]);
+
+        $disk = Storage::disk('gridfs-prefix');
+        $disk->put('hello/world.txt', 'Hello World!');
```

Comment:
I may be mistaken and/or misunderstanding something, but aren't all files supposed to be prefixed with a leading directory separator?

Reply:
I recall that the PathPrefixer adds it, but I thought the docs might present the canonical form. I just noted the absence of a leading `/` is consistent with examples in the [Laravel Filesystem docs](https://laravel.com/docs/11.x/filesystem#deleting-files).

Reply:
When the prefix is empty, the file name in the `fs.files` collection does not have the leading `/`. The only situation you can get a leading `/` is by having it in the prefix.